strawberry/union.py

```diff
@@ -254,9 +254,36 @@ def union(
     ... class A: ...
     >>> @strawberry.type
     ... class B: ...
-    >>> strawberry.union("Name", (A, Optional[B]))
+    >>> Annotated[A | B, strawberry.union("Name")]
     """
 
+    if types is None:
+        _source_file = None
+        _source_line = None
+
+        if should_use_rich_exceptions():
+            frame = sys._getframe(1)
+
+            _source_file = frame.f_code.co_filename
+            _source_line = frame.f_lineno
+
+        return StrawberryUnion(
+            name=name,
+            description=description,
+            directives=directives,
+            _source_file=_source_file,
+            _source_line=_source_line,
+        )
+
+    warnings.warn(
+        (
+            "Passing types to `strawberry.union` is deprecated. Please use "
+            f'{name} = Annotated[Union[A, B], strawberry.union("{name}")] instead'
+        ),
+        DeprecationWarning,
+        stacklevel=2,
+    )
```

Comment:
How about reducing one nested lvl here by putting thr `warnings` in a `if types is not None` and with an early return, and the main code with no indentation?

Reply:
it's pretty much the same 😊 since the rest of the code will go in the if :)